[GCC 12.2.0] build fails with: error: return type 'struct tm' is incomplete

The build of ES 2.10.3 fails since GCC 12.2.0 because of:

```
FAILED: es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o 
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-g++ -DUSE_OPENGLES_20 -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/freetype2 -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/SDL2 -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/vlc -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/external -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src -march=armv8-a+crc -mtune=cortex-a53 -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -mfloat-abi=hard -mfpu=neon-fp-armv8 -Wall -pipe  -flto -fno-fat-lto-objects -O2 -fomit-frame-pointer -DNDEBUG -std=c++11 -Wno-attributes -O2 -O3 -DNDEBUG -MD -MT es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o -MF es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o.d -o es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o -c /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.h:5,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/utils/TimeUtil.h:41:37: error: field 'mTimeStruct' has incomplete type 'tm'
   41 |                         tm          mTimeStruct;
      |                                     ^~~~~~~~~~~
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/cwchar:44,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/bits/postypes.h:40,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/bits/char_traits.h:39,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/string:40,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/utils/TimeUtil.h:5:
```

Looks like this is faulty: https://github.com/RetroPie/EmulationStation/blob/v2.10.3/es-core/src/utils/TimeUtil.h#L41

### full build log
```
BUILD      emulationstation (target)
    TOOLCHAIN      cmake (auto-detect)
Executing (target): cmake -GNinja -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DCMAKE_TOOLCHAIN_FILE=/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/etc/cmake-armv8a-libreelec-linux-gnueabihf.conf -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release -DGL=off -DGLES=on -DUSE_MESA_GLES=on /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Deprecation Warning at CMakeLists.txt:16 (cmake_policy):
  The OLD behavior for policy CMP0072 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found GLESv2: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libGLESv2.so
-- Found Freetype: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libfreetype.so (found version "2.12.1") 
-- Looking for FreeImage...
-- Found PkgConfig: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/bin/pkg-config (found version "0.29.2") 
-- Checking for module 'freeimage'
--   No package 'freeimage' found
-- Found FreeImage: optimized;/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libfreeimage.so;debug;/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libfreeimage.so
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found SDL2: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libSDL2main.a;/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libSDL2.so  
-- Found SDL2_mixer: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libSDL2_mixer.so (found version "2.6.2") 
-- Found CURL: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libcurl.so (found version "7.84.0")  
-- Checking for module 'libvlc>=1.0.0'
--   Found libvlc, version 3.0.17.3
-- VLC library found
-- Found VLC: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/vlc  
CMake Warning (dev) at /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/share/cmake-3.24/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (Rapidjson)
  does not match the name of the calling package (RapidJSON).  This can lead
  to problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  CMake/Packages/FindRapidJSON.cmake:64 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  CMakeLists.txt:96 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Rapidjson: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include  
-- Found ALSA: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/lib/libasound.so (found version "1.2.7.2") 
CMake Deprecation Warning at external/pugixml/CMakeLists.txt:3 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Configuring done
-- Generating done
-- Build files have been written to: /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/.armv8a-libreelec-linux-gnueabihf
Executing (target): ninja 
[5/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/AudioManager.cpp.o
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/AudioManager.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/AudioManager.h: In constructor 'AudioManager::AudioManager()':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/AudioManager.h:27:14: warning: 'AudioManager::mInitialized' will be initialized after [-Wreorder]
   27 |         bool mInitialized;
      |              ^~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/AudioManager.h:22:20: warning:   'Mix_Music* AudioManager::mCurrentMusic' [-Wreorder]
   22 |         Mix_Music* mCurrentMusic;
      |                    ^~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/AudioManager.cpp:14:1: warning:   when initialized here [-Wreorder]
   14 | AudioManager::AudioManager() : mInitialized(false), mCurrentMusic(nullptr), mMusicVolume(Settings::getInstance()->getInt("MusicVolume"))
      | ^~~~~~~~~~~~
[7/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/GuiComponent.cpp.o
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.h: In constructor 'GuiComponent::GuiComponent(Window*)':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.h:148:23: warning: 'GuiComponent::mParent' will be initialized after [-Wreorder]
  148 |         GuiComponent* mParent;
      |                       ^~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.h:145:23: warning:   'unsigned char GuiComponent::mOpacity' [-Wreorder]
  145 |         unsigned char mOpacity;
      |                       ^~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:11:1: warning:   when initialized here [-Wreorder]
   11 | GuiComponent::GuiComponent(Window* window) : mWindow(window), mParent(NULL), mOpacity(255),
      | ^~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.h:169:23: warning: 'GuiComponent::mTransform' will be initialized after [-Wreorder]
  169 |         Transform4x4f mTransform; //Don't access this directly! Use getTransform()!
      |                       ^~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.h:162:14: warning:   'bool GuiComponent::mIsProcessing' [-Wreorder]
  162 |         bool mIsProcessing;
      |              ^~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:11:1: warning:   when initialized here [-Wreorder]
   11 | GuiComponent::GuiComponent(Window* window) : mWindow(window), mParent(NULL), mOpacity(255),
      | ^~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp: In member function 'bool GuiComponent::finishAnimation(unsigned char)':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:348:28: warning: unused variable 'done' [-Wunused-variable]
  348 |                 const bool done = mAnimationMap[slot]->update(mAnimationMap[slot]->getAnimation()->getDuration() - mAnimationMap[slot]->getTime());
      |                            ^~~~
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:7:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = Vector2f; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:417:46:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  120 |                         else if(std::is_same<T, unsigned int>::value) return *(const T*)&properties.at(prop).i;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  122 |                         else if(std::is_same<T, bool>::value)         return *(const T*)&properties.at(prop).b;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  123 |                         else if(std::is_same<T, Vector4f>::value)     return *(const T*)&properties.at(prop).r;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = float; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:430:39:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  120 |                         else if(std::is_same<T, unsigned int>::value) return *(const T*)&properties.at(prop).i;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  122 |                         else if(std::is_same<T, bool>::value)         return *(const T*)&properties.at(prop).b;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = bool; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/GuiComponent.cpp:441:29:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:118:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  118 |                         if(     std::is_same<T, Vector2f>::value)     return *(const T*)&properties.at(prop).v;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  120 |                         else if(std::is_same<T, unsigned int>::value) return *(const T*)&properties.at(prop).i;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:121:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  121 |                         else if(std::is_same<T, float>::value)        return *(const T*)&properties.at(prop).f;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  123 |                         else if(std::is_same<T, Vector4f>::value)     return *(const T*)&properties.at(prop).r;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[8/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/HelpStyle.cpp.o
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/resources/Font.h:9,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HelpStyle.cpp:3:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = Vector2f; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HelpStyle.cpp:25:33:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  120 |                         else if(std::is_same<T, unsigned int>::value) return *(const T*)&properties.at(prop).i;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  122 |                         else if(std::is_same<T, bool>::value)         return *(const T*)&properties.at(prop).b;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  123 |                         else if(std::is_same<T, Vector4f>::value)     return *(const T*)&properties.at(prop).r;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = unsigned int; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HelpStyle.cpp:31:38:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:118:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  118 |                         if(     std::is_same<T, Vector2f>::value)     return *(const T*)&properties.at(prop).v;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:121:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  121 |                         else if(std::is_same<T, float>::value)        return *(const T*)&properties.at(prop).f;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  122 |                         else if(std::is_same<T, bool>::value)         return *(const T*)&properties.at(prop).b;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  123 |                         else if(std::is_same<T, Vector4f>::value)     return *(const T*)&properties.at(prop).r;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[9/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/HttpReq.cpp.o
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.h: In constructor 'HttpReq::HttpReq(const std::string&)':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.h:66:16: warning: 'HttpReq::mStatus' will be initialized after [-Wreorder]
   66 |         Status mStatus;
      |                ^~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.h:64:15: warning:   'CURL* HttpReq::mHandle' [-Wreorder]
   64 |         CURL* mHandle;
      |               ^~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.cpp:40:1: warning:   when initialized here [-Wreorder]
   40 | HttpReq::HttpReq(const std::string& url)
      | ^~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.cpp: In destructor 'HttpReq::~HttpReq()':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/HttpReq.cpp:126:19: warning: suggest explicit braces to avoid ambiguous 'else' [-Wdangling-else]
  126 |                 if(merr != CURLM_OK)
      |                   ^
[11/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/InputConfig.cpp.o
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/InputConfig.cpp: In member function 'void InputConfig::loadFromXML(pugi::xml_node&)':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/InputConfig.cpp:188:19: warning: suggest explicit braces to avoid ambiguous 'else' [-Wdangling-else]
  188 |                 if(value == 0)
      |                   ^
[15/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/platform.cpp.o
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/platform.cpp: In function 'void processQuitMode()':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/platform.cpp:72:16: warning: enumeration value 'QUIT' not handled in switch [-Wswitch]
   72 |         switch (quitMode)
      |                ^
[21/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/Window.cpp.o
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h: In constructor 'Window::Window()':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h:103:14: warning: 'Window::mNormalizeNextUpdate' will be initialized after [-Wreorder]
  103 |         bool mNormalizeNextUpdate;
      |              ^~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h:97:13: warning:   'int Window::mFrameTimeElapsed' [-Wreorder]
   97 |         int mFrameTimeElapsed;
      |             ^~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.cpp:18:1: warning:   when initialized here [-Wreorder]
   18 | Window::Window() : mNormalizeNextUpdate(false), mFrameTimeElapsed(0), mFrameCountElapsed(0), mAverageDeltaTime(10),
      | ^~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h:107:22: warning: 'Window::mTimeSinceLastInput' will be initialized after [-Wreorder]
  107 |         unsigned int mTimeSinceLastInput;
      |                      ^~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h:89:25: warning:   'Window::ScreenSaver* Window::mScreenSaver' [-Wreorder]
   89 |         ScreenSaver*    mScreenSaver;
      |                         ^~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.cpp:18:1: warning:   when initialized here [-Wreorder]
   18 | Window::Window() : mNormalizeNextUpdate(false), mFrameTimeElapsed(0), mFrameCountElapsed(0), mAverageDeltaTime(10),
      | ^~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h:91:25: warning: 'Window::mRenderScreenSaver' will be initialized after [-Wreorder]
   91 |         bool            mRenderScreenSaver;
      |                         ^~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.h:90:25: warning:   'Window::InfoPopup* Window::mInfoPopup' [-Wreorder]
   90 |         InfoPopup*      mInfoPopup;
      |                         ^~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/Window.cpp:18:1: warning:   when initialized here [-Wreorder]
   18 | Window::Window() : mNormalizeNextUpdate(false), mFrameTimeElapsed(0), mFrameCountElapsed(0), mAverageDeltaTime(10),
      | ^~~~~~
[22/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/animations/AnimationController.cpp.o
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/animations/AnimationController.cpp: In destructor 'virtual AnimationController::~AnimationController()':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/animations/AnimationController.cpp:15:9: warning: deleting object of abstract class type 'Animation' which has non-virtual destructor will cause undefined behavior [-Wdelete-non-virtual-dtor]
   15 |         delete mAnimation;
      |         ^~~~~~~~~~~~~~~~~
[25/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/components/ButtonComponent.cpp.o
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/ButtonComponent.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/ButtonComponent.h: In constructor 'ButtonComponent::ButtonComponent(Window*, const std::string&, const std::string&, const std::function<void()>&)':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/ButtonComponent.h:48:28: warning: 'ButtonComponent::mBox' will be initialized after [-Wreorder]
   48 |         NinePatchComponent mBox;
      |                            ^~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/ButtonComponent.h:34:31: warning:   'std::shared_ptr<Font> ButtonComponent::mFont' [-Wreorder]
   34 |         std::shared_ptr<Font> mFont;
      |                               ^~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/ButtonComponent.cpp:6:1: warning:   when initialized here [-Wreorder]
    6 | ButtonComponent::ButtonComponent(Window* window, const std::string& text, const std::string& helpText, const std::function<void()>& func) : GuiComponent(window),
      | ^~~~~~~~~~~~~~~
[28/112] Building CXX object es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o
FAILED: es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o 
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/bin/armv8a-libreelec-linux-gnueabihf-g++ -DUSE_OPENGLES_20 -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/freetype2 -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/SDL2 -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/vlc -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/external -I/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src -march=armv8-a+crc -mtune=cortex-a53 -mabi=aapcs-linux -Wno-psabi -Wa,-mno-warn-deprecated -D_FILE_OFFSET_BITS=64 -D_TIME_BITS=64 -mfloat-abi=hard -mfpu=neon-fp-armv8 -Wall -pipe  -flto -fno-fat-lto-objects -O2 -fomit-frame-pointer -DNDEBUG -std=c++11 -Wno-attributes -O2 -O3 -DNDEBUG -MD -MT es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o -MF es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o.d -o es-core/CMakeFiles/es-core.dir/src/components/DateTimeComponent.cpp.o -c /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.h:5,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp:1:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/utils/TimeUtil.h:41:37: error: field 'mTimeStruct' has incomplete type 'tm'
   41 |                         tm          mTimeStruct;
      |                                     ^~~~~~~~~~~
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/cwchar:44,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/bits/postypes.h:40,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/bits/char_traits.h:39,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/include/c++/12.2.0/string:40,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/utils/TimeUtil.h:5:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/toolchain/armv8a-libreelec-linux-gnueabihf/sysroot/usr/include/wchar.h:83:8: note: forward declaration of 'struct tm'
   83 | struct tm;
      |        ^~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/utils/TimeUtil.h:28:87: error: return type 'struct tm' is incomplete
   28 |                                    operator tm         ()                       const { return mTimeStruct; }
      |                                                                                       ^
In file included from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/resources/Font.h:9,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/TextComponent.h:5,
                 from /build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.h:6:
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = bool; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp:95:37:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:118:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  118 |                         if(     std::is_same<T, Vector2f>::value)     return *(const T*)&properties.at(prop).v;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  120 |                         else if(std::is_same<T, unsigned int>::value) return *(const T*)&properties.at(prop).i;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:121:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  121 |                         else if(std::is_same<T, float>::value)        return *(const T*)&properties.at(prop).f;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  123 |                         else if(std::is_same<T, Vector4f>::value)     return *(const T*)&properties.at(prop).r;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = unsigned int; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp:101:35:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:118:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  118 |                         if(     std::is_same<T, Vector2f>::value)     return *(const T*)&properties.at(prop).v;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:121:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  121 |                         else if(std::is_same<T, float>::value)        return *(const T*)&properties.at(prop).f;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  122 |                         else if(std::is_same<T, bool>::value)         return *(const T*)&properties.at(prop).b;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:123:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  123 |                         else if(std::is_same<T, Vector4f>::value)     return *(const T*)&properties.at(prop).r;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h: In instantiation of 'const T ThemeData::ThemeElement::get(const std::string&) const [with T = float; std::string = std::__cxx11::basic_string<char>]':
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/components/DateTimeComponent.cpp:126:34:   required from here
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:120:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  120 |                         else if(std::is_same<T, unsigned int>::value) return *(const T*)&properties.at(prop).i;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/ThemeData.h:122:79: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
  122 |                         else if(std::is_same<T, bool>::value)         return *(const T*)&properties.at(prop).b;
      |                                                                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/build/LibreELEC-RR/build.LibreELEC-AMLGX.arm-11.0-devel/build/emulationstation-3d22c283d8081eea755dd6ef6c2ed4a7db5d6c09/es-core/src/utils/TimeUtil.h:11:28: warning: 'Utils::Time::NOT_A_DATE_TIME' defined but not used [-Wunused-variable]
   11 |                 static int NOT_A_DATE_TIME = 0;
      |                            ^~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
FAILURE: scripts/build emulationstation during make_target (default)
*********** FAILED COMMAND ***********
ninja ${NINJA_OPTS} ${PKG_MAKE_OPTS_TARGET}
**************************************
```

--- a/es-core/src/utils/TimeUtil.h
+++ b/es-core/src/utils/TimeUtil.h
@@ -3,6 +3,7 @@
 #define ES_CORE_UTILS_TIME_UTIL_H
 
 #include <string>
+#include <time.h>
 
 namespace Utils
 {

